src/store_swapout.cc

```diff
@@ -369,6 +372,14 @@ StoreEntry::mayStartSwapOut()
         return false;
     }
 
+    // in SMP mode, restrict caching to StoreEntry publisher to avoid
+    // workers releasing each other caching attempts
+    if (!Store::Root().transientsWriter(*this)) {
```

Comment:
Don't we need to limit the scope of this condition to SMP cases, as the comment prescribes, e.g.,
```suggestion
    if (!Store::Root()::SmpAware && !Store::Root().transientsWriter(*this)) {
```

Also, I found another problem: our unit tests (testRock and testUfs) are broken after ddc7d7e. The reason is in the following contradiction: though testRock is 'SmpAware', it does not have Transients created due to stub UsingSmp(), returning false.

Reply:
> unit tests (testRock and testUfs) are broken

Glad you noticed. I probably did not run `make check`. Please fix.

Does any existing unit test create Transients, StoreMap, or any similar complex shared memory structures?

Reply:
> Don't we need to limit the scope of this condition to SMP cases, as the comment prescribes

Yes, we do! The suggested change looks backwards to me though. I would

```C++
if (Store::Root()::SmpAware && !Store::Root().transientsWriter(*this)) {
```

Earlier versions checked `transientsReader()` here, but that felt too weak. I lost the (previously implied) SMP check when converting to !transientsWriter(). Root()::SmpAware() is equivalent to Controller having Transients, right?

Reply:
Done at 4443485.